### DISCUSSION TITLE
Can fd sort by nearest match?

### DISCUSSION THREAD

**Discussion by jamischarles** [sharkdp/fd#1267] [Q&A] [answered]
I'm trying to find the nearest file matching a pattern. 

**Use case:**
Given a test file (in neovim) I'm trying to find the associated code file (and vice versa). I'd love to not hardcode for relative paths etc. ie: `util.ts` -> `./__test__/util.test.ts`. and `util.test.ts` -> `../util.ts`

Does fd have any sense of startPath and resultPath? So we can measure the distance between the two folders? I'm trying to narrow down to the result with the shortest distance between paths that matches the pattern.

**Comment by tavianator** [comment, marked as answer]
There's nothing like that implemented right now, but see https://github.com/sharkdp/fd/issues/734 for some similar discussion
